### PR TITLE
VMEncryption: For confidential VM temp disk enc, do not attempt secure key release if KEK is not provided

### PR DESCRIPTION
**Pull request by canfikret**
This is to unblock tdx CVM until GA is fixed.

### CODE REVIEW

File: VMEncryption/main/handle.py [hunk 1]
```diff
@@ -797,7 +797,9 @@ def enable():
         encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
         check_release_rsa_or_ec_key = False
         if security_Type == CommonVariables.ConfidentialVM:
-            check_release_rsa_or_ec_key = True
+            if CommonVariables.KeyEncryptionKeyURLKey in public_settings:
```
Comment (pankajosh) [inline VMEncryption/main/handle.py:800]: are we saying KEK is not required if temp disk encryption is attempted, whereas if data disk encryption is attempted then KEK is required and SKR must be attempted.

Comment (canfikret) [reply VMEncryption/main/handle.py:800]: Correct.

Comment (pankajosh) [reply VMEncryption/main/handle.py:800]: currently I do not see the precheck for 
ConfidentialEncryptionDataDisk is true then KEK should be present, 
or we can include this check this condition for SKR. 
if ConfidentialEncryptionDataDisk and KEK 
    do SKR

Comment (canfikret) [reply VMEncryption/main/handle.py:800]: We can add that check for ConfidentialEncryptionDataDisk true scenario. However, note that this flag will go away when data disk enc goes public preview.

Comment (pankajosh) [reply VMEncryption/main/handle.py:800]: yes, I agree with your comments.

**Review by pankajosh — APPROVED** [review]

**Pull request merged**